central/sensorupgradeconfig/datastore/singleton.go

```diff
@@ -32,12 +32,11 @@ func addDefaultConfigIfEmpty(d DataStore) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to check initial sensor upgrade config")
 	}
-	if currentConfig != nil && (!env.ManagedCentral.BooleanSetting() || !currentConfig.GetEnableAutoUpgrade()) {
+	if currentConfig != nil {
 		return nil
 	}
 
-	// Auto upgrade is disabled by default if managed central flag is set
-	return d.UpsertSensorUpgradeConfig(ctx, upgradeConfig(!env.ManagedCentral.BooleanSetting()))
+	return d.UpsertSensorUpgradeConfig(ctx, upgradeConfig(true))
```

Comment:
Oh, I dropped the ball here, it definitely shouldn't be defaulted to `true`.

However, I'm a bit uneasy with setting this in the `GetSensorUpgradeConfig` thought. It still feels like it should be the behavior of the datastore startup. Another issue is that the service context comes with the user's scoped access. Which might not allow write/updates in the DB. Resulting in a code like this:

```go
func (s *service) insertDefaultSensorConfig(ctx context.Context) (*storage.SensorUpgradeConfig, error) {
	// This is needed in case the first `GET` request to SensorUpgradeConfig is done by a user
	// without permission to add sensor configs. We still want to create the default
	// config entry.
	allAccessContext := sac.WithAllAccess(ctx)
	config := defaultUpgradeConfig()
	if err := s.configDataStore.UpsertSensorUpgradeConfig(allAccessContext, config); err != nil {
		return nil, err
	}
	return config, nil
}
``` 
Pushed a commit with this. Take a look and let me know what do you think.

Reply:
My main point is: why persist (insert into the database) the default value at all? If all requests go through Service, the Service's `GetSensorUpgradeConfig()` can just return the `defaultUpgradeConfig()` when DataStore returns `nil`.

Same logic for `autoTrigger`, set it according to what `GetSensorUpgradeConfig()` returns upon Service startup.

If you still want to insert the default value in the database, do it through `GetSensorUpgradeConfig()` but construct all-mighty `Context` before calling that function.